packages/react-components/react-color-picker-preview/library/docs/Spec.md

```diff
@@ -2,62 +2,458 @@
+
+export const Default = () => {
+  const COLOR = 'rgba(0, 255, 170, 1)';
+  const [selectedColor, setSelectedColor] = React.useState(COLOR);
+  const handleChange: ColorPickerOnSelectEventHandler = (_, data) => {
+    setSelectedColor(data.selectedColorHex);
+  };
+
+  return (
+    <>
+      <ColorPicker color={selectedColor} onChange={handleChange}>
+        <ColorArea />
+        <AlphaSlider />
+        <HueSlider />
+      </ColorPicker>
+      <div style={{ backgroundColor: `${selectedColor}` }} />
+    </>
+  );
+};
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+### Slider variants
+
+- horizontal (default)
+- vertical
+
+ColorSliders might represent different color channels.
+
+### Shapes
+
+- `square` (default)
+- `rounded`
+
+### Size
+
+Size will be the same as in Office products.
+For custom sizes users might want to customize it via CSS.
+
+### States
+
+- `rest`
+- `focused`
+
+### Color support
+
+- RGB(A)
+- Hex
+- HSL(A)
+- HSV/HSB
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### ColorPicker
+
+| Property   | Values              | Default   | Purpose                                |
+| ---------- | ------------------- | --------- | -------------------------------------- |
+| color      | `string`            |           | Sets color value                       |
+| onChange   | `function`          | undefined | Callback called when color is selected |
+| shape      | `square`, `rounded` | `square`  | Sets shape                             |
+| step       | `number`            | 1         | Step for arrow navigation              |
+| customStep | `number`            | 3         | Fast navigation                        |
+
+Input fields with color values will be in `renderUtils`. It will contain default ColorPicker which has all colors represented and a preview swatch. This functionality is similar to V8.
+
+```ts
+import { ColorPickerLayout } from '@fluentui/react-color-picker-preview';
+```
+
+```tsx
+export const Default = () => {
+  const COLOR = 'rgba(0, 255, 170, 1)';
+  const [selectedColor, setSelectedColor] = React.useState(COLOR);
+  const handleChange: ColorPickerOnSelectEventHandler = (_, data) => {
+    setSelectedColor(data.selectedColorHex);
+  };
+
+  return (
+    <>
+      <ColorPickerLayout
+        color={selectedColor}
+        onChange={handleChange}
+        showAlphaSlider={true}
+        showPreview={true}
+        channels={['hex', 'rgb']}
+      />
+      <div style={{ backgroundColor: `${selectedColor}` }} />
+    </>
+  );
+};
+```
+
+![ColorPickerLayout](./assets/color-picker-layout.jpg)
+
+Validation of color fields will be in `utils`. Validation should not allow entering incorrect values to the input fields.
+
+For color manipulation and conversion `tinycolor` library will be used.
```

Comment:
I hope that not [this one](https://www.npmjs.com/package/tinycolor) 😸 Can you please reference a NPM package instead?

Reply:
It's [this one](https://www.npmjs.com/package/@ctrl/tinycolor) and it's already installed in our repo